hmpps-sqs-spring-boot-autoconfigure/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsNoDlqQueueFactoryTest.kt

```diff
@@ -16,285 +15,115 @@ import org.mockito.kotlin.verifyNoMoreInteractions
     }
   }
 
   @Nested
-  inner class `Create single LocalStack HmppsQueue with no dlq` {
+  inner class `Create LocalStack HmppsQueue with asynchronous SQS clients` {
     private val someQueueConfig = QueueConfig(queueName = "some queue name")
     private val hmppsSqsProperties = HmppsSqsProperties(provider = "localstack", queues = mapOf("somequeueid" to someQueueConfig))
-    private val sqsClient = mock<SqsClient>()
+    private val sqsAsyncClient = mock<SqsAsyncClient>()
     private lateinit var hmppsQueues: List<HmppsQueue>
 
     @BeforeEach
     fun `configure mocks and register queues`() {
-      whenever(sqsFactory.localstackSqsClient(anyString(), anyString()))
-        .thenReturn(sqsClient)
-      whenever(sqsClient.getQueueUrl(any<GetQueueUrlRequest>())).thenReturn(GetQueueUrlResponse.builder().queueUrl("some queue url").build())
+      whenever(sqsClientFactory.localstackSqsAsyncClient(anyString(), anyString()))
+        .thenReturn(sqsAsyncClient)
+      whenever(sqsAsyncClient.getQueueUrl(GetQueueUrlRequest.builder().queueName("some queue name").build()))
+        .thenReturn(CompletableFuture.completedFuture(GetQueueUrlResponse.builder().queueUrl("some queue url").build()))
+      whenever(sqsAsyncClient.createQueue(any<CreateQueueRequest>()))
+        .thenReturn(CompletableFuture.completedFuture(CreateQueueResponse.builder().build()))
 
       hmppsQueues = hmppsQueueFactory.createHmppsQueues(hmppsSqsProperties)
     }
 
     @Test
-    fun `creates LocalStack sqs client from sqs factory but not dlq client`() {
-      verify(sqsFactory).localstackSqsClient(localstackUrl = "http://localhost:4566", region = "eu-west-2")
-      verifyNoMoreInteractions(sqsFactory)
+    fun `should create async client but no dlq from sqs factory`() {
+      verify(sqsClientFactory).localstackSqsAsyncClient("http://localhost:4566", "eu-west-2")
+      verifyNoMoreInteractions(sqsClientFactory)
     }
 
     @Test
-    fun `should return the queue details`() {
-      assertThat(hmppsQueues[0].id).isEqualTo("somequeueid")
-    }
-
-    @Test
-    fun `should return the queue SqsClient client`() {
-      assertThat(hmppsQueues[0].sqsClient).isEqualTo(sqsClient)
-    }
-
-    @Test
-    fun `should return the queue name`() {
-      assertThat(hmppsQueues[0].queueName).isEqualTo("some queue name")
-    }
-
-    @Test
-    fun `should return the queue url`() {
-      assertThat(hmppsQueues[0].queueUrl).isEqualTo("some queue url")
-    }
-
-    @Test
-    fun `should not return a dlq client`() {
+    fun `should return async client but no dlq client`() {
+      assertThat(hmppsQueues[0].sqsClient).isInstanceOf(SqsAsyncClient::class.java)
       assertThat(hmppsQueues[0].sqsDlqClient).isNull()
     }
 
     @Test
-    fun `should not return a dlq name`() {
-      assertThat(hmppsQueues[0].dlqName).isNull()
-    }
-
-    @Test
-    fun `should not return a dlq url`() {
-      assertThat(hmppsQueues[0].dlqUrl).isNull()
```

Comment:
This NoDlqAsync test class misses a lot of tests that are in the non-async test class (such as multiple queues etc). I wonder if it would be better to convert the non-async tests to async?